Tools/scripts/build_ci.sh

```diff
@@ -365,10 +365,18 @@ for t in $CI_BUILD_TARGET; do
 
     if [ "$t" == "replay" ]; then
         echo "Building replay"
-        $waf configure --board sitl --debug --disable-scripting
+
+        if [ $cxx_compiler == "g++" ]; then
+            # Only be strict about Werror on SITL when compiling with g++
+            # Once the warnings are gone in clang, remove this exception
+            maybe_werror="--Werror"
+        fi
+        $waf configure --board sitl --debug --disable-scripting $maybe_werror
+        
         $waf replay
         echo "Building AP_DAL standalone test"
-        $waf configure --board sitl --debug --disable-scripting --no-gcs
+        $waf configure --board sitl --debug --disable-scripting --no-gcs $maybe_werror
```

Comment:
i'd prefer the whitelist active for normal SITL by default, and add a --disable-Werror configure option

Reply:
It's using `optparse`, the precursor to `argparse`. I'll figure it out.

Reply:
Waf enables Werror by default in 30c354dde4a7385d6530cfb689d2f8fff4d24fd8

Reply:
And, it's now enabled in SITL, same as ChibiOS, in 9b89d26a0c0257ed528e08b1c3967e6ee032b251